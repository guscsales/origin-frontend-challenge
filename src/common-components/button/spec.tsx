import * as React from 'react';
import { mount } from 'enzyme';
import Button from '.';
import theme from '../../configs/styles/themes';
import { ThemeProvider } from 'styled-components';
import { create } from 'react-test-renderer';

describe('Button', () => {
  describe('#render', () => {
    const getWrapper = (props: any = {}, renderType: any = mount) =>
      renderType(
        <ThemeProvider theme={theme}>
          <Button {...props}>{props.children}</Button>
        </ThemeProvider>
      );

    it('should return a `button`', () => {
      const wrapper = getWrapper({ children: 'test' });

      expect(wrapper.find('button')).toHaveLength(1);
    });

    it('should match a snapshot', () => {
      const wrapper = getWrapper({ children: 'test' }, create).toJSON();

      expect(wrapper).toMatchSnapshot();
    });
  });
});
