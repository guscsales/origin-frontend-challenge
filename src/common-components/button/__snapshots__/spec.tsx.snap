// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button #render should match a snapshot 1`] = `
.c0 {
  width: 100%;
  color: #fff;
  font-size: 1.8rem;
  border-radius: 30px;
  border: none;
  font-weight: 500;
  padding: 1.5rem;
  margin-top: 3.5rem;
  cursor: pointer;
  -webkit-transition: all 200ms linear;
  transition: all 200ms linear;
  background-color: #1b31a8;
  font-family: Work Sans;
}

.c0:hover {
  background-color: #192d98;
}

@media (min-width:1024px) {
  .c0 {
    max-width: 40rem;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    margin: 3rem auto 1.5rem;
  }
}

<button
  className="c0"
>
  test
</button>
`;
